# EDA Dataset

## Explore

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns   

In [2]:
list1 =np.array(range(10))
list1

array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [9]:
df_ebay = pd.read_csv('../data/raw/ebayShort.csv', index_col=0)

As the data is too big, we first working in the shortcut of the total data with 1 million rows.

In [10]:
# Look at the first 5 row

df_ebay.head()

b2c_c2c  seller_id  declared_handling_days      acceptance_scan_timestamp  \
0     B2C      25454                     3.0  2019-03-26 15:11:00.000-07:00   
1     C2C    6727381                     2.0  2018-06-02 12:53:00.000-07:00   
2     B2C      18507                     1.0  2019-01-07 16:22:00.000-05:00   
3     B2C       4677                     1.0  2018-12-17 16:56:00.000-08:00   
4     B2C       4677                     1.0  2018-07-27 16:48:00.000-07:00   

   shipment_method_id  shipping_fee  carrier_min_estimate  \
0                   0           0.0                     3   
1                   0           3.0                     3   
2                   0           4.5                     3   
3                   0           0.0                     3   
4                   0           0.0                     3   

   carrier_max_estimate    item_zip buyer_zip  category_id  item_price  \
0                     5       97219     49040           13       27.95   
1                     5  11415-3528     62521            0       20.50   
2                     5       27292     53010            1       19.90   
3                     5       90703     80022            1       35.50   
4                     5       90703     55070            1       25.00   

   quantity               payment_datetime delivery_date  weight  \
0         1  2019-03-24 03:56:49.000-07:00    2019-03-29       5   
1         1  2018-06-01 13:43:54.000-07:00    2018-06-05       0   
2         1  2019-01-06 00:02:00.000-05:00    2019-01-10       9   
3         1  2018-12-16 10:28:28.000-08:00    2018-12-21       8   
4         1  2018-07-26 18:20:02.000-07:00    2018-07-30       3   

   weight_units            package_size  record_number  
0             1                  LETTER              1  
1             1  PACKAGE_THICK_ENVELOPE              2  
2             1  PACKAGE_THICK_ENVELOPE              3  
3             1  PACKAGE_THICK_ENVELOPE              4  
4             1  PACKAGE_THICK_ENVELOPE              5

In [12]:
#Look at 5 random row
df_ebay.sample(5)

b2c_c2c  seller_id  declared_handling_days  \
361117     C2C     207874                     3.0   
649560     B2C      94638                     2.0   
738778     B2C       3397                     1.0   
48042      B2C       4343                     5.0   
186585     B2C        685                     0.0   

            acceptance_scan_timestamp  shipment_method_id  shipping_fee  \
361117  2019-07-31 11:53:00.000-04:00                   0           5.0   
649560  2019-04-10 21:49:00.000-05:00                   0           0.0   
738778  2019-07-22 16:29:00.000-04:00                   0           0.0   
48042   2018-03-20 16:58:00.000-05:00                   0           0.0   
186585  2018-05-07 15:12:00.000-07:00                   0           0.0   

        carrier_min_estimate  carrier_max_estimate item_zip buyer_zip  \
361117                     3                     5    27576     56377   
649560                     3                     5    77025     15217   
738778                     3                     5    08701     94804   
48042                      3                     5    76210     30273   
186585                     3                     5    95826     18106   

        category_id  item_price  quantity               payment_datetime  \
361117            2        5.00         1  2019-07-30 20:13:39.000-04:00   
649560            7       19.00         4  2019-04-09 21:55:54.000-05:00   
738778           11       39.99         1  2019-07-21 12:13:10.000-04:00   
48042             0       16.00         1  2018-03-12 20:04:27.000-05:00   
186585           12       10.95         1  2018-05-04 12:09:52.000-07:00   

       delivery_date  weight  weight_units            package_size  \
361117    2019-08-03       0             1  PACKAGE_THICK_ENVELOPE   
649560    2019-04-12      12             1  PACKAGE_THICK_ENVELOPE   
738778    2019-07-24      16             1  PACKAGE_THICK_ENVELOPE   
48042     2018-03-26       0             1                    NONE   
186585    2018-05-09       0             1  PACKAGE_THICK_ENVELOPE   

        record_number  
361117         361118  
649560         649561  
738778         738779  
48042           48043  
186585         186586

In [15]:
#check the shape of the dataset
df_ebay.shape

(1000000, 19)

In [16]:
print(f'There are {df_ebay.shape[0]} columns and {df_ebay.shape[1]} rows')

There are 1000000 columns and 19 rows


In [17]:
# Get a quick overview of dataset variables
df_ebay.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1000000 entries, 0 to 999999
Data columns (total 19 columns):
 #   Column                     Non-Null Count    Dtype  
---  ------                     --------------    -----  
 0   b2c_c2c                    1000000 non-null  object 
 1   seller_id                  1000000 non-null  int64  
 2   declared_handling_days     952820 non-null   float64
 3   acceptance_scan_timestamp  1000000 non-null  object 
 4   shipment_method_id         1000000 non-null  int64  
 5   shipping_fee               1000000 non-null  float64
 6   carrier_min_estimate       1000000 non-null  int64  
 7   carrier_max_estimate       1000000 non-null  int64  
 8   item_zip                   1000000 non-null  object 
 9   buyer_zip                  1000000 non-null  object 
 10  category_id                1000000 non-null  int64  
 11  item_price                 1000000 non-null  float64
 12  quantity                   1000000 non-null  int64  
 13  payment_datet

In [18]:
# top 5 rows showing only 'object' columns
df_ebay.select_dtypes('object').head()

b2c_c2c      acceptance_scan_timestamp    item_zip buyer_zip  \
0     B2C  2019-03-26 15:11:00.000-07:00       97219     49040   
1     C2C  2018-06-02 12:53:00.000-07:00  11415-3528     62521   
2     B2C  2019-01-07 16:22:00.000-05:00       27292     53010   
3     B2C  2018-12-17 16:56:00.000-08:00       90703     80022   
4     B2C  2018-07-27 16:48:00.000-07:00       90703     55070   

                payment_datetime delivery_date            package_size  
0  2019-03-24 03:56:49.000-07:00    2019-03-29                  LETTER  
1  2018-06-01 13:43:54.000-07:00    2018-06-05  PACKAGE_THICK_ENVELOPE  
2  2019-01-06 00:02:00.000-05:00    2019-01-10  PACKAGE_THICK_ENVELOPE  
3  2018-12-16 10:28:28.000-08:00    2018-12-21  PACKAGE_THICK_ENVELOPE  
4  2018-07-26 18:20:02.000-07:00    2018-07-30  PACKAGE_THICK_ENVELOPE

## Data Cleaning

First let do some common checking in the dataset.
 -  Null value
 -  Duplicate row

In [19]:
#Checking null
df_ebay.isna().sum()

b2c_c2c                          0
seller_id                        0
declared_handling_days       47180
acceptance_scan_timestamp        0
shipment_method_id               0
shipping_fee                     0
carrier_min_estimate             0
carrier_max_estimate             0
item_zip                         0
buyer_zip                        0
category_id                      0
item_price                       0
quantity                         0
payment_datetime                 0
delivery_date                    0
weight                           0
weight_units                     0
package_size                     0
record_number                    0
dtype: int64

There is 47180 missing value in total 1 million rows. We could delete it but I decide to have deepper look at it later


In [24]:
# checking duplicate rows
df_ebay.duplicated().any()

False

There are no duplicate row in the database